starlite/config.py

```diff
@@ -33,9 +34,47 @@ class CORSConfig(BaseModel):
     max_age: int = 600
 
 
-class GZIPConfig(BaseModel):
+class CompressionBackend(str, Enum):
+    GZIP = "gzip"
+    BROTLI = "brotli"
+
+
+class BrotliMode(str, Enum):
+    GENERIC = "generic"
+    TEXT = "text"
+    FONT = "font"
+
+
+class CompressionConfig(BaseModel):
+    backend: Optional[CompressionBackend] = None
     minimum_size: int = 500
-    compresslevel: int = 9
+    gzip_compress_level: int = 9
+    brotli_quality: int = 5
+    brotli_mode: BrotliMode = BrotliMode.TEXT
+    brotli_lgwin: int = 22
+    brotli_lgblock: int = 0
+    brotli_gzip_fallback: bool = True
+    CompressionMiddleware: PyObject = None
```

Comment:
So this part and the init method I'd dump. I would instead use a `property` that has this logic OR I would move this logic into the consumer.